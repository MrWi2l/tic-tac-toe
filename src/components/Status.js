import React from 'react'
import calculateWinner from '../calculateWinner'

const StatusContainer = ({ squares, currentPlayer }) => {
  let status
  let state
  const winner = calculateWinner(squares)

  if (squares.filter(x => x !== 'X' && x !== 'O').length === 0) {
    status = 'DRAW'
    state = 'draw'
  } else if (winner) {
    status = `Winner is : ${winner}`
    state = 'winner'
  } else {
    status = `Next player : ${currentPlayer}`
  }
  return <div className={state}>{status}</div>
}

StatusContainer.defaultProps = {
  currentPlayer: '',
}

StatusContainer.propTypes = {
  squares: React.PropTypes.array.isRequired,
  currentPlayer: React.PropTypes.string.isRequired,
}

module.exports = StatusContainer
