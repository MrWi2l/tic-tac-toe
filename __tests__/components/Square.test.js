import React from 'react'
import { shallow } from 'enzyme'
import Status from '../../src/components/Square'

describe('<Square />', () => {
  it('Render X', () => {
    const wrapper = shallow(
      <Status
        value={'X'}
        onClick={() => null}
      />,
    )
    expect(wrapper.text()).toMatchSnapshot()
  })
})
