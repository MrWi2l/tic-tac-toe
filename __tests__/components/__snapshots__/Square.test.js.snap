exports[`<Square /> Render X 1`] = `"X"`;
